src/datetime/Time.php

```diff
@@ -0,0 +1,481 @@
+   * -1 if shorter, and 0 if they are equal.
+   */
+  public function compare(this $other): int {
+    if ($this->hours !== $other->hours) {
+      return $this->hours <=> $other->hours;
+    }
+    if ($this->minutes !== $other->minutes) {
+      return $this->minutes <=> $other->minutes;
+    }
+    if ($this->seconds !== $other->seconds) {
+      return $this->seconds <=> $other->seconds;
+    }
+    return $this->nanoseconds <=> $other->nanoseconds;
+  }
+
+  public function isEqual(this $other): bool {
+    return $this->compare($other) === 0;
+  }
+
+  public function isShorter(this $other): bool {
+    return $this->compare($other) === -1;
+  }
+
+  public function isShorterOrEqual(this $other): bool {
+    return $this->compare($other) <= 0;
+  }
+
+  public function isLonger(this $other): bool {
+    return $this->compare($other) === 1;
+  }
+
+  public function isLongerOrEqual(this $other): bool {
+    return $this->compare($other) >= 0;
+  }
+
+  /**
+   * Returns true if this instance represents a time interval longer than $a but
+   * shorter than $b, or vice-versa (shorter than $a but longer than $b), or if
+   * this instance is equal to $a and/or $b. Returns false if this instance is
+   * shorter/longer than both.
+   */
+  public function isBetweenIncl(this $a, this $b): bool {
+    $a = $this->compare($a);
+    $b = $this->compare($b);
+    return $a === 0 || $a !== $b;
+  }
+
+  /**
+   * Returns true if this instance represents a time interval longer than $a but
+   * shorter than $b, or vice-versa (shorter than $a but longer than $b).
+   * Returns false if this instance is equal to $a and/or $b, or shorter/longer
+   * than both.
+   */
+  public function isBetweenExcl(this $a, this $b): bool {
+    $a = $this->compare($a);
+    $b = $this->compare($b);
+    return $a !== 0 && $b !== 0 && $a !== $b;
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // operations
+
+  /**
+   * Returns a new instance, converting a positive/negative interval to the
+   * opposite (negative/positive) interval of equal length. The resulting
+   * instance has all parts equivalent to the current instance's parts
+   * multiplied by -1.
+   */
+  public function invert(): this {
+    if ($this->isZero()) {
+      return $this;
+    }
+    return new self(
+      -$this->hours,
+      -$this->minutes,
+      -$this->seconds,
+      -$this->nanoseconds,
+    );
+  }
+
+  /**
+   * Returns a new instance representing the sum of this instance and the
+   * provided `$other` instance. Note that time intervals can be negative, so
+   * the resulting instance is not guaranteed to be shorter/longer than either
+   * of the inputs.
+   *
+   * This operation is commutative: `$a->plus($b) === $b->plus($a)`
+   */
+  public function plus(this $other): this {
+    if ($other->isZero()) {
+      return $this;
+    } else if ($this->isZero()) {
+      return $other;
+    }
+    return self::hours(
```

Comment:
name seems misleading if it can take minutes, seconds, ad nanoseconds, should be "create" or "fromPrimitives"

Reply:
Good feedback! I wasn't 100% sure about this naming pattern, but my general idea was to optimize for the common case (which is a single argument  like `Time::hours(42)`), then having the option to provide the additional arguments is just a bonus on top of that (users can also choose to avoid it if they find it confusing I guess).

In my native language it's also reasonable to write something like "3:30 hours" which is understood to mean "3 hours and 30 minutes", but I'm not sure if this applies to English as well :)

`create`/`fromPrimitives`/`fromParts` (or just having a public constructor) could be a good replacement for the current version of `hours` (maybe we can support both? though I have mixed feelings about having too many ways to do the same thing) -- but I think what's nice about the current version is that there is a nice way to create an instance with any combinations of parts.

Reply:
personally, i would prefer: `Time\Interval::empty()->withDays(3)->withHours(12)->withMinutes(32)->withSeconds(31)`

also, Tokio time is a good reference here: https://docs.rs/tokio/1.6.1/tokio/time/index.html

Reply:
That's...very verbose, I don't think I would want to write that regularly, but it works with the current version so you have that option too if you prefer it (performance might be a bit of a concern though, since if we want to keep the objects immutable, that would have to create multiple throwaway intermediate objects).

Reply:
![image](https://user-images.githubusercontent.com/360927/124008390-fc40cf80-d9a1-11eb-8e04-329d979edbb7.png)

Though perhaps the optional args should go from all of the current ones? I'm don't think `::minutes(12, 30, 100)` is clear when reading either